### PR TITLE
Enhancement#12/test environment

### LINKED ISSUE #12: setup fo unit test

### PR DESCRIPTION
**Pull request by juliocabrera820**
# Description

I just set up the unit test environment
I added some gems such as: factory bot, faker,database cleaner and rspec-rails
With the configuration we can use new matchers to test our models and create fake data in the next unit test tasks

closes #12 

I've added support for new matchers and fake data, as well as cleaning the test database after completing a unit test

### CODE REVIEW

**Review by YilKanda — APPROVED** [review]
LGTM!

**Review by YilKanda — APPROVED** [review]
LGTM!

As a recommendation, try to give a detailed description of what and why was done in both the issue and the PR. Since I observe that your issue appears empty.

**Comment by juliocabrera820** [comment]
> LGTM!
> 
> As a recommendation, try to give a detailed description of what and why was done in both the issue and the PR. Since I observe that your issue appears empty.

You're right,I did not fill in the issue description

**Review by jairolan — APPROVED** [review]
LGTM

File: reloj-checador/Gemfile [hunk 1]
```diff
@@ -49,10 +49,15 @@ group :test do
   gem 'capybara', '>= 3.26'
   gem 'selenium-webdriver'
   # Easy installation and use of web drivers to run system tests with browsers
+  gem 'database_cleaner-active_record'
```
Comment (anxhe) [inline reloj-checador/Gemfile:52]: Only as a comment, this gem isn't necessary now that rails cleans the data created during each test by default via transactions test, https://guides.rubyonrails.org/testing.html#testing-parallel-transactions

File: reloj-checador/spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
+
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+  config.after(:all) do
```
Comment (anxhe) [inline reloj-checador/spec/rails_helper.rb:64]: same as above

**Review by anxhe — COMMENTED** [review]
Great set for testing 🚀

File: reloj-checador/Gemfile (hunk 1 above)
Comment (juliocabrera820) [reply reloj-checador/Gemfile:52]: I'll consider this

File: reloj-checador/spec/rails_helper.rb (hunk 2 above)
Comment (juliocabrera820) [reply reloj-checador/spec/rails_helper.rb:64]: I'll consider this

**Pull request merged**